File: leetcode/problems_list/128-LongestConsecutiveSequence/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
  const numSet = new Set(nums);
  let longest = 0;

  numSet.forEach((e) => {
    if (!numSet.has(e - 1)) {
      let end = e + 1;
      while (numSet.has(end)) {
        end += 1;
        longest = longest > end - e ? longest : end - e;
      }
    }
  });

  return longest === 0 && numSet.size > 0 ? 1 : longest;
};

File: leetcode/problems_list/128-LongestConsecutiveSequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        longest = 0
        for num in num_set:
            if num - 1 not in num_set:
                end = num + 1
                while end in num_set:
                    end += 1
                    longest = max(longest, end - num)
        return 1 if (longest == 0 and len(num_set) > 0) else longest
